packages/styled-components/src/models/StyledComponent.js

```diff
@@ -253,10 +240,11 @@ function useStyledComponentImpl<Config: {}, Instance>(
       props.className, // eslint-disable-line react/prop-types
       styledComponentId,
       attrs.className,
-      generatedClassName
+      generatedClassName !== styledComponentId ? generatedClassName : null
```

Comment:
for a static style use case the generated name seems to just be the static component id, so without this the className is added twice